Bugzilla/Template.pm

```diff
@@ -260,6 +260,129 @@ sub quoteUrls {
     return $text;
 }
 
+sub quoteUrls2 {
+    my ($text, $bug, $comment) = @_;
+    return $text unless $text;
+    my $anon_user = new Bugzilla::User;
```

Comment:
Instead of forking get_bug_link and get_attachment_link, I pass an anonymous user to those methods. This will make this quoteUrls2 consistent for caching while at the same time preventing duplication of those methods.